tests/gui/test_DirectSpinBox.py

```diff
@@ -0,0 +1,117 @@
+from direct.showbase.ShowBase import ShowBase
+from direct.showbase import ShowBaseGlobal
+from direct.gui.DirectSpinBox import DirectSpinBox
+from direct.gui import DirectGuiGlobals as DGG
+import pytest
+
+from direct.showbase.Loader import Loader
+import pytest
+import sys
+if sys.version_info >= (3, 0):
+    import builtins
+else:
+    import __builtin__ as builtins
+builtins.loader = Loader(base=None)
+
+from direct.task.TaskManagerGlobal import taskMgr
+builtins.taskMgr = taskMgr
+from direct.showbase.MessengerGlobal import messenger
+builtins.messenger = messenger
```

Comment:
I would rather change DirectSpinBox to rely on as few builtins as possible.  This can be done by adding the following to `DirectSpinBox.py`:
```
from direct.showbase.MessengerGlobal import messenger
from direct.task.TaskManagerGlobal import taskMgr
```

I realise this may not be possible (yet) for Loader.

Reply:
Makes sense, will change the class according to your proposal